packages/server/lib/video_capture.ts

```diff
@@ -274,32 +275,39 @@ export function start (name, options: StartOptions = {}) {
 type OnProgress = (p: number) => void
 
 export async function process (name, cname, videoCompression, ffmpegchaptersConfig, onProgress: OnProgress = function () {}) {
-  const metaFileName = `${name}.meta`
-
-  const maybeGenerateMetaFile = Bluebird.method(() => {
-    if (!ffmpegchaptersConfig) {
-      return false
-    }
-
-    // Writing the metadata to filesystem is necessary because fluent-ffmpeg is just a wrapper of ffmpeg command.
-    return fs.writeFile(metaFileName, ffmpegchaptersConfig).then(() => true)
-  })
-
-  const addChaptersMeta = await maybeGenerateMetaFile()
-
   let total = null
 
+  const metaFileName = `${name}.meta`
+  const addChaptersMeta = ffmpegchaptersConfig && await fs.writeFile(metaFileName, ffmpegchaptersConfig).then(() => true)
+
   return new Bluebird((resolve, reject) => {
     debug('processing video from %s to %s video compression %o',
       name, cname, videoCompression)
 
     const command = ffmpeg()
+    .addOptions([
+      // These flags all serve to reduce initial buffering, especially important
+      // when dealing with very short videos (such as during component tests)
+      '-avioflags direct',
+      '-fpsprobesize 0',
+      '-probesize 32',
+      '-analyzeduration 0',
+
+    ])
+
     const outputOptions = [
       '-preset fast',
-        `-crf ${videoCompression}`,
-        '-pix_fmt yuv420p',
+      `-crf ${videoCompression}`,
+      '-pix_fmt yuv420p',
     ]
 
+    // On low-resource machines - estimated here by CPU count - limit the encoder
+    // to a single thread. While this can slow down video processing, it
+    // significantly reduces CPU contention, leading to a faster overall test time.
+    if (os.cpus().length < 4) {
+      outputOptions.push('-threads 1')
+    }
```

Comment:
Good idea. Would it be a good idea to always upper-bound this? 

Like `-threads ${os.cpus().length / 2}` or something?

Reply:
So after some discussion in Slack, I'm going to try *always* limiting it to one thread, because `os.cpus().length` doesn't account for docker, which uses cgroups to limit resource usage per-container but doesn't hide the actual cpu count.

Multithreading the encoder is useful when you're not limited by input speed - reading a file off the disk or network, faster than a single thread can process it, rather than passing in frames live at a relatively low framerate (as we do).

Reply:
👍

Reply:
I appear to have been incorrect; Changing to a single encoding thread added 60 seconds to each build node, more than counteracting the 20s saved by changing the probe options. Going to remove this arg entirely.